packages/messenger-widget/src/components/Contacts/bl.ts

```diff
@@ -192,9 +195,13 @@ export const setContactSelectedFromCache = (
     state: GlobalState,
     cacheContacts: ContactPreview[],
 ): number | null => {
-    const name = state.accounts.selectedContact?.account.ensName;
+    const key =
+        state.accounts.selectedContact?.account.profile?.publicEncryptionKey;
     for (let index = 0; index < cacheContacts.length; index++) {
```

Comment:
If there is no good reason to use for() please use a higher order function. I think .filter() would work for that case

Reply:
Ok, will replace for loop with findIndex high order function as I needed index here.